src/main/webapp/js/lib/calendar.js

```diff
@@ -206,6 +243,19 @@ function createDateFromTimeString(time, day) {
  */
 function clear() {
   calendar.clear(/*immediately=*/ true);
+  // Duplicate the originalColors array and rerandomize the potential schedule
+  // colors.
+  scheduleColors = [...originalColors];
+  randomizeColors();
+}
+
+/**
+ * Randomizes the scheduleColors array.
+ */
+function randomizeColors() {
+  scheduleColors.sort(() => {
+    return 0.5 - Math.random();
```

Comment:
Highly recommend you read this article https://medium.com/@nitinpatel_20236/how-to-shuffle-correctly-shuffle-an-array-in-javascript-15ea3f84bfb (and the [supporting analysis article](http://www.robweir.com/blog/2010/02/microsoft-random-browser-ballot.html)) for why this is not a good approach.